### ISSUE TITLE
Workflow: Rebase docs on develop upon push to develop

### ISSUE THREAD

**Issue by vlipovac** [pmgbergen/porepy#796]
The idea is to use simple console instructions in a workflow `.yml`-file to perform a standard git rebase by checking out the
repo in a temporary directory and perform respective operations inside.

If it works, we lose a dependency on third-part actions.

**Comment by vlipovac** [comment]
I suggest the following course of action

- [ ]  1.  A workflow/action to ensure no folder `/docds/` is contained in a PR to develop
- [ ]  2. Another action asserting that PR to docs contains **only** modifications in folder `/docs/`
- [ ]  3. A manual synchronization of `docs` with `develop`, shortly before the last point.
- [ ]  4. Introduce Action which rebases `docs` onto `develop` once a PR-Merger is performed.

General idea to write actions with console commands doing above with standard git commands.
Mind the authorization inside each workflow (Token?) since both branches are protected.

**Comment by keileg** [comment]
@vlipovac Do I understand you correctly that point 3 is a one-time manual operation while the rest is automatized? If so, this sounds great.

**Comment by vlipovac** [comment]
> @vlipovac Do I understand you correctly that point 3 is a one-time manual operation while the rest is automatized? If so, this sounds great.

Yes, step 3 is a one-time manual operation, to be done right before the workflow from 4 is introduced (including a coordination that no PR to `develop` is performed while steps 3-4 are conducted).

**Comment by vlipovac** [comment]
Put on hold for now, until documentation of the package is brought to an acceptable level.

References for how to approach this:
https://github.com/ammaraskar/sphinx-action
